gtk4/src/pad_action_entry.rs

```diff
@@ -21,32 +19,34 @@ impl PadActionEntry {
         action_name: &str,
     ) -> Self {
         assert_initialized_main_thread!();
-        Self {
-            type_,
-            index,
-            mode,
-            label: label.to_owned(),
-            action_name: action_name.to_owned(),
+        unsafe {
+            Self::unsafe_from(ffi::GtkPadActionEntry {
+                type_: type_.into_glib(),
+                index,
+                mode,
+                label: label.as_ptr() as *const _,
+                action_name: action_name.as_ptr() as *const _,
```

Comment:
This borrows the strings without constraining things by lifetime. You should use `to_glib_full()` here and then you also need a `Drop` impl that frees the strings (or rather: add the corresponding functions into the macro above).